src/main/java/mil/dds/anet/threads/mart/services/MartReportImporterService.java

```diff
@@ -0,0 +1,402 @@
+
+@Component
+public class MartReportImporterService implements IMartReportImporterService {
+
+  private static final String SECURITY_MARKING_JSON_PROPERTY = "securityMarking";
+  protected final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+  public static final String REPORT_JSON_ATTACHMENT = "mart_report.json";
+
+  private final ObjectMapper ignoringMapper = MapperUtils.getDefaultMapper()
+      .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+      .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
+
+  private final OrganizationDao organizationDao;
+  private final ReportDao reportDao;
+  private final TaskDao taskDao;
+  private final PersonDao personDao;
+  private final PositionDao positionDao;
+  private final LocationDao locationDao;
+  private final MartImportedReportDao martImportedReportDao;
+  private final AttachmentDao attachmentDao;
+  private final EmailAddressDao emailAddressDao;
+
+  public MartReportImporterService(ReportDao reportDao, PersonDao personDao,
+      PositionDao positionDao, TaskDao taskDao, OrganizationDao organizationDao,
+      LocationDao locationDao, MartImportedReportDao martImportedReportDao,
+      AttachmentDao attachmentDao, EmailAddressDao emailAddressDao) {
+    this.reportDao = reportDao;
+    this.personDao = personDao;
+    this.positionDao = positionDao;
+    this.taskDao = taskDao;
+    this.organizationDao = organizationDao;
+    this.locationDao = locationDao;
+    this.martImportedReportDao = martImportedReportDao;
+    this.attachmentDao = attachmentDao;
+    this.emailAddressDao = emailAddressDao;
+  }
+
+  @Override
+  public void processMartReport(List<FileAttachment> attachments) {
+    Optional<FileAttachment> martReportAttachmentOpt = attachments.stream()
+        .filter(attachment -> attachment.getName().equalsIgnoreCase(REPORT_JSON_ATTACHMENT))
+        .findFirst();
+
+    // Do we have a MART report in this email?
+    if (martReportAttachmentOpt.isPresent()) {
+      ReportDto reportDto;
+
+      // Get the report JSON from the attachment
+      FileAttachment martReportAttachment = martReportAttachmentOpt.get();
+      try {
+        martReportAttachment.load();
+        reportDto =
+            getReportInfo(new String(martReportAttachment.getContent(), StandardCharsets.UTF_8));
+
+        MartImportedReport newMartImportedReport = new MartImportedReport();
+        newMartImportedReport.setSequence(reportDto.getSequence());
+        newMartImportedReport.setReportUuid(reportDto.getUuid());
+        newMartImportedReport.setSubmittedAt(reportDto.getSubmittedAt());
+        newMartImportedReport.setReceivedAt(Instant.now());
+
+        // First check, is this report uuid in MartImportedReports table already?
+        MartImportedReport existingMartImportedReport =
+            martImportedReportDao.getByReportUuid(reportDto.getUuid());
+
+        if (existingMartImportedReport != null) {
```

Comment:
Suggest to rewrite this `if` to avoid the "dangling else":
```java
        if (existingMartImportedReport == null) {
          // New report, import
          processReportInfo(reportDto, newMartImportedReport, attachments);
          martImportedReportDao.insert(newMartImportedReport);
        } else if (existingMartImportedReport.isSuccess()) {
          // If it was successfully imported it is a duplicate, do not import again
          […]
        } else if (!existingMartImportedReport.isSuccess() && Objects.equals(
            existingMartImportedReport.getSequence(), newMartImportedReport.getSequence())) {
          // This report was marked as failed or missing earlier, import now and replace the
          // existing martImportedReport
          […]
        }
```